aten/src/ATen/cuda/cub.cu

```diff
@@ -42,8 +51,12 @@ struct CountMaskOp {
 
 void mask_exclusive_sum(const uint8_t *mask, int64_t *output_idx, int64_t n) {
   CountMaskOp op{};
+#if CUB_V3_PLUS()
+  auto iter = thrust::transform_iterator<decltype(op), decltype(mask)>(mask, op);
+#else
   auto iter = NO_ROCM(at_cuda_detail)::cub::TransformInputIterator<
       bool, decltype(op), decltype(mask)>(mask, op);
+#endif
```

Comment:
Ditto, I believe this can be applied unconditionally. Also we can leverage `thrust::make_transform_iterator`
```suggestion
  auto iter = thrust::make_transform_iterator(mask, op);
```

Reply:
NO_ROCM wrappers are there for a reason, to enable rocm builds, that said it's true that it would be better to enable this with a macro similar to NO_ROCM instead of ifdefs